include/openthread/platform/infra_if.h

```diff
@@ -138,33 +138,6 @@ extern void otPlatInfraIfRecvIcmp6Nd(otInstance         *aInstance,
  */
 extern otError otPlatInfraIfStateChanged(otInstance *aInstance, uint32_t aInfraIfIndex, bool aIsRunning);
 
-/**
- * Send a request to discover the NAT64 prefix on the infrastructure interface with @p aInfraIfIndex.
- *
- * OpenThread will call this method periodically to monitor the presence or change of NAT64 prefix.
- *
- * @param[in]  aInfraIfIndex  The index of the infrastructure interface to discover the NAT64 prefix.
- *
- * @retval  OT_ERROR_NONE    Successfully request NAT64 prefix discovery.
- * @retval  OT_ERROR_FAILED  Failed to request NAT64 prefix discovery.
- */
-otError otPlatInfraIfDiscoverNat64Prefix(uint32_t aInfraIfIndex);
-
-/**
- * The infra interface driver calls this method to notify OpenThread that
- * the discovery of NAT64 prefix is done.
- *
- * Is expected to be invoked after calling otPlatInfraIfDiscoverNat64Prefix.
- * If no NAT64 prefix is discovered, @p aIp6Prefix shall point to an empty prefix with zero length.
- *
- * @param[in]  aInstance      The OpenThread instance structure.
- * @param[in]  aInfraIfIndex  The index of the infrastructure interface on which the NAT64 prefix is discovered.
- * @param[in]  aIp6Prefix     A pointer to NAT64 prefix.
- */
-extern void otPlatInfraIfDiscoverNat64PrefixDone(otInstance        *aInstance,
```

Comment:
On Android we're depending on this API to notify OT core about the discovered NAT64 prefix. The current Android implementation is, the platform discovers the NAT64 via PREF64 RA option and we set the prefix using this `otPlatInfraIfDiscoverNat64PrefixDone` API.

https://cs.android.com/android/platform/superproject/main/+/main:external/ot-br-posix/src/android/android_rcp_host.cpp;l=372;drc=61197364367c9e404c7da6900658f1b16c42d0da

Reply:
OK. But requiring OT to periodically poll for PREF64 information seems backward. As noted in RFC 8781, one reason for using PREF64s in RAs is that the information and any updates are actively pushed from the router.

Is it possible to remove `otPlatInfraIfDiscoverNat64Prefix()`?

Reply:
I reverted most of the changes for now. But I'd like to eventually remove the periodic timer that polls the platform for the NAT64 prefix.

Reply:
I just checked the Android code. It should be fine to fully remove the `otPlatInfraIfDiscoverNat64Prefix(uint32_t aInfraIfIndex)` and the periodic timer. On Android platform it's the platform side to notify OT core about the discovered NAT64 prefix, triggered by the infra link's events. 

https://cs.android.com/android/platform/superproject/main/+/main:packages/modules/Connectivity/thread/service/java/com/android/server/thread/ThreadNetworkControllerService.java;l=868;drc=6303c84ef593d2f898e154667bc61b61120416eb